src/Desktop/UIAutomation/CustomObjects/CustomProperty.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using System;
+
+namespace Axe.Windows.Desktop.UIAutomation.CustomObjects
+{
+    public class CustomProperty
+    {
+        /// <summary>The RFC4122 globally unique identifier of this property.</summary>
+        [JsonProperty("guid")]
+        #pragma warning disable CA1720 // Identifier contains type name: name from JSON
+        public Guid? Guid { get; set; }
```

Comment:
Why is this property optional? Later we check to ensure it isn't null.

Reply:
Guid is a value type, so it can't be checked for null unless it's nullable (the `?` operator). We could use a non-nullable Guid and check for Guid.Empty in the validation step if that's preferable.

Reply:
I like Guid.empty. It seems more straight forward to me.